src/utils/getDefaultDockerContext.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import { DockerOptions } from 'dockerode';
+import * as util from 'util';
+
+const exec = util.promisify(cp.exec);
+
+export async function getDefaultDockerContext(): Promise<DockerOptions | undefined> {
+    try {
+        const { stdout, stderr } = await exec('docker context inspect');

```

Comment:
Doesn't this assume the Docker CLI is installed (which seems a little weird to do in order to configure Dockerode, whose purpose is not need the Docker CLI)?  At the very least, can we catch the case where it isn't installed.  (I know it's an unlikely scenario, but I'd rather not unnecessarily add to our dependencies on the Docker CLI.)

Also, should we be checking the exit code as well as stderr?

Reply:
If it's not installed we'll fall through the catch and down to the old behavior.

Reply:
I'm going to rename it though to `tryGetDefaultDockerContext` since it's strictly a best-effort thing.